speechbrain/utils/metric_stats.py

```diff
@@ -112,7 +116,7 @@ def append(self, ids, *args, **kwargs):
                 scores = multiprocess_evaluation(
                     metric=self.metric, n_jobs=self.n_jobs, **kwargs
                 )
-
+        scores = sb.utils.distributed_metrics.gather_for_metrics(scores)
```

Comment:
I thought we had agreed that `gather`-ing in the `summarize` method was better than in `append`

Reply:
I think 'better' is the issue. Yes, it is performance-wise. However, the issue is that not all metrics have a `summarize` function, like for instance `def minDCF`. Furthermore, what would happen if the user needs to access internal states of metrics now? If they never run the `summarize` function, then it means that the internal state will only be for the current rank process. When doing unit testing, I found it particularly useful to access internal states before running `summarize`. I think doing sync in `append` is the most straightforward way to sync metrics and also the most natural way for an end-user.

I also took a look at other toolkits and found that they were also synchronizing during the `append` stage.

Reply:
Okay, you've convinced me I guess.